policy-design-lab/pdl-frontend
# #329: Add selector to EQIP and CSP total page

This PR addresses issues #329 and #331.

## Changes Made
1. Selector Operation Added: The Selector operation has been added to both the EQIP and CSP pages.
2. Column Pagination Implemented: Column pagination has been created for their tables to prevent overcrowding when users add many practices. This ensures the table does not expand excessively, preserving the overall layout.
3. IRA Updates: The selector tables in IRA have been updated to incorporate this new feature.


## How to Test

1. Navigate to the EQIP page on the development site: https://policydesignlab-dev.ncsa.illinois.edu/eqip. You should see the default status reflect "All Practices" for both the map and the table:
<img width="1280" alt="Screenshot 2024-11-21 at 4 16 33 PM" src="https://github.com/user-attachments/assets/929ef8b6-8444-410c-ad04-bb7853e49360">

2. Select random practices. Test with more than three practices to observe column pagination. YThe number displayed in the map area should change based on your selections:
<img width="1305" alt="Screenshot 2024-11-21 at 4 18 04 PM" src="https://github.com/user-attachments/assets/607189b6-65af-4d77-8373-335539931ebd">


3. Check that the table at the bottom of the page reflects your selections.Column pagination should function correctly when more than six columns are displayed:
<img width="1469" alt="Screenshot 2024-11-21 at 4 18 53 PM" src="https://github.com/user-attachments/assets/ea8a7bb5-d5c0-4bd4-9b01-e2b7d9f8cde1">
<img width="1387" alt="Screenshot 2024-11-21 at 4 19 32 PM" src="https://github.com/user-attachments/assets/5dc61cec-2a7c-4ace-bd20-0e9c4e0303f5">

4. Click the "Export this table to csv" button and you should be able to see the full table. Verify the table data.

5. Repeart step 1 to 4 for the CSP: https://policydesignlab-dev.ncsa.illinois.edu/csp

6. Play with IRA tables at https://policydesignlab-dev.ncsa.illinois.edu/ira to verify the column pagniation feature.
